### PR TITLE
ab: course setup show form errors

### PR DESCRIPTION
**Pull request by annagav**
Course setup: show form validation errors. 
Fixed a bug in providing 'based_on' assignment.

### CODE REVIEW

File: instructor/common.py [hunk 1]
```diff
@@ -97,19 +97,28 @@ def course_setup(request):
             # Create a new course
             user = User.objects.get(username=request.user)
             form = CourseForm(request.POST)
-            course = form.save(commit=False)
-            course.owner_id = user.id
-            course.save()
+            if form.is_valid():
+                course = form.save(commit=False)
+                course.owner_id = user.id
+                course.save()
```
Comment (giocalitri) [inline instructor/common.py:103]: question: do you need this to be in a transaction?

File: instructor/common.py [hunk 2]
```diff
@@ -97,19 +97,28 @@ def course_setup(request):
...
+            else:
+                all_courses = models.Course.objects.all()
+                return render_to_response('instructor/course_setup.html',
+                                          {'courses': all_courses,
+                                           'form': form,
+                                           'new': request.session['new']},
+                                          context_instance=RequestContext(request))
```
Comment (giocalitri) [inline instructor/common.py:110]: you have to indent this better

File: instructor/common.py (hunk 1 above)
Comment (annagav) [inline instructor/common.py:103]: I don't think I understand the question.

Comment (giocalitri) [inline instructor/common.py:103]: fine

**Pull request merged**